### PR TITLE
Ports GenPop System from NSV13

### PR DESCRIPTION
**Pull request by maxion12345**
# Github documenting your Pull Request
This adds the General Population system to Yogurtstation13! Included in a new brig interface with preset crimes, and a new way to keep prisoners secured!
Heres a photo of the interface.

As a demonstration, I will brig Brodie Brinigh for Theft and Trespass. We'll input the details, and it sets the time.

Hitting "Finalize ID" prints out the completed prisoner ID, and begins tracking our time in the brig. It also prints a piece of paperwork to be filed by the warden (or thrown away). The ID looks something like this:

Next, we'll need to strip our prisoner. Find an un-assigned GenPop locker, open it, and strip the prisoner of everything but his jumpsuit (ID included)

To seal the locker, slide the ID over the locker and it locks shut and assigns itself to that ID. It can be opened by the prisoner later once the brig time is up. Then just put the prisoner into a cell with the temporary ID, and you are good to go! The system automatically updates arrest status and security records.

Once released, special access is added to the prisoner ID card that enables them to walk out of turnstiles, so they can collect their stuff and let themselves out. This feature isn't pictured as turnstiles haven't been added into the game yet.

NEW MAPS (turnstiles to be added later)
Yogurtstation: Upgraded cells with genpop

Meta: has basically the same layout as yogurtstation

Kilostation: just changed the terminals and lockers to gen
pop and added flasher button under em

Delta: Only two cells so no room for genpop

# Wiki Documentation

Update brig photos
Update brigging procedure to brig with this system instead of the old one
New procedure:
1. Take the prisoner to the brig and inform the Warden of their crimes so their Security Record may be updated.
2. Take the prisoner to a Genpop Interface, input crimes, and print the ID.
3. Find an empty genpop locker, open the locker and hold the prisoner over it.
4. Empty their pockets and remove their gloves, backpacks, tool belt, gas masks, and any flash resistant equipment such as Sunglasses, Welding Masks/Goggles, Space Helmets, etc. Make sure the prisoner is clothed once you have completed this step, but ensure to remove their ID.
5. Search the items removed and be sure to check the internals box in their backpack.
6. Confiscate any contraband and/or stolen items, as well as any tools that may be used for future crimes.
These are to be placed in evidence, not left on the brig floor or your personal use, until they can be returned to their rightful owners.
7. Close the locker and lock it by swiping the Genpop ID over it. 
8. Put ID onto prisoner, and place them within a cell.
# Changelog

:cl:  NSV13 & Maxim
rscadd: Added the General Population brigging system, complete with automatic time setting and group cell capabilities.
/:cl:

### CODE REVIEW

**Comment by maxion12345** [comment]
whoop fixed it all we're golden now

**Comment by Chubbygummibear** [comment]
This is rad, love seeing slick and intuitive UIs with tgui.

**Comment by Tipy1802** [comment]
Conflicted on this, while it does seem helpful, wouldn't it put the warden out of a job?

**Comment by maxion12345** [comment]
did it work

**Comment by maxion12345** [comment]
this is hell man all i wanted to do was fetch upstream

**Comment by maxion12345** [comment]
ok we are back where we started
fuck

**Comment by maxion12345** [comment]
new plan
we don't touch maps until turnstiles are merged and we can do it all at once because every time I fetch upstream it cuts yogstation in half and causes massive merge conflicts

**Comment by maxion12345** [comment]
WE ARE GO FOR GENPOP

**Comment by maxion12345** [comment]
@wejengin2 
not sure taking it out of draft pings for review so here i am

**Review by patpol4 — APPROVED** [review]
Didnt go into a detailed review but it seems fine on the surface, you might want to pixelshift the basketball hoop later.
 Most issues will probably come out in testmerge.

**Comment by maxion12345** [comment]
speaking of testmerge, im going to try to refine the UI a bit more. Currently there is an issue with someone spamming crimes, so im going to try to fix this by having the "current crimes" put off to the side of the timer in a box. I do not have the dev server so this will be pain to make. It IS functional right now if you want to testmerge just for a concept type of deal, but its not very refined.

**Comment by maxion12345** [comment]
actually nevermind it wasn't that hard I just had to use my brain

**Comment by burn1cycle** [comment]
We already have lots of people who break space law and not so many people who want to play sec so why are we building a feature that rewards people for breaking the law while handicapping people trying to enforce it?


This is what you get for giving people nice things and adding three extra steps to brigging procedures just for them:
[Round 37302]
> Den Ward tries to remove Daniel Daringer's handcuffs.
> Shhhhhh facepalms
> Daniel Daringer throws the handcuffs.
> Den Ward puts the handcuffs into the security backpack.
> **Daniel Daringer has slashed Den Ward! x2**(<<<-----------------------------------------------)
> (then a little bit later the mime helped)
> Shhhhhh digs into the heated security screws and starts removing them...
> Shhhhhh wedges the pocket crowbar into the gap in the frame and starts prying...
> Shhhhhh starts cutting the exposed bars on the reinforced window...
> Shhhhhh starts unfastening the reinforced window from the frame...
> (and then Daniel and Shhhhh ran off to freedom never to be seen again)


You should've been aware that BeeStation(origin of feature) has a much more significant set of rules to govern a feature like this https://nsv.beestation13.com/wiki/Rules
> 1. Protocol & Orders. You are an officer of your faction, act like one. Take orders from people of higher rank than you. Follow protocol when available and prudent. If you think the person giving you the order is being unreasonable, find an IC to deal with it, and if necessary open an admin ticket.
> 2. Lethal Force. Do not physically attack any other crewmember without a legitimate, explainable, roleplay reason that could be applied in a similar, real-life scenario. Always call security and never use lethal intent as an opening against own crewmembers. Nanotrasen do use lethal force when absolutely necessary. Executions are extremely rare but are the final resort. You must ahelp before an official execution takes place. Any crew members injured or killed in crew-on-crew combat must be revived if able.
> 3. (irrelevant)
> 4. Mutiny. Mutiny is not allowed unless an Admin clears it via Adminhelp. You must also follow Mutiny Policy.

**Comment by maxion12345** [comment]
time to go fix random conflict

**Comment by maxion12345** [comment]
reverted maps so I can pull master to fix conflicts, re-making them now. I'll also take this time to look into that bug.

File: tgui/packages/tgui/interfaces/GenPop.js
Comment (JamieD1) [inline tgui/packages/tgui/interfaces/GenPop.js]:
```suggestion
```

**Comment by maxion12345** [comment]
Development of this will be stopped for a few days, I'll be away from spaceman's. The test merge has been informative so far, and I've noted somethings for further development, thank you for any feedback

**Comment by burn1cycle** [comment]
Feature has been a disaster. 
 There's no safe way to put people in genpop when it's full or if the prisoners are violent it's also dangerous. What I've observed is security avoiding brigging to avoid having to use genpop. I have yet to see a command or sec regular whose enthused about it however it's getting positive reviews from shitters who break spacelaw and ruin the round for others.

**Comment by maxion12345** [comment]
after gathering some feedback from players, I believe the main issues of this system revolve around the big gen pop cell. Namely:
1) Getting people into the cell is a pain when there are others already inside
2) Prisoners are assholes and beat each other up instead of just coloring, playing arcade, and balling
3) due to the nature of it being a cell, it's difficult to actually break up these fights

Most feedback thinks the new brig system is actually good, and it all falls apart when it comes time to put the prisoners into a cell without prisoners. With that in mind, I am going to be removing the big general cell, and instead of having only one cell per prisoner, as it was before. Delta and Kilo already use this, if you want an example of how it'll look. I'll take this time to spruce up the brig cells as well, because I still want the prisoners to have something to do, even if it's just read space law. maybe I'll give them a shitter.

**Comment by maxion12345** [comment]
taking the time to merge master so there are no conflicts later down the line

**Comment by maxion12345** [comment]
figured out that the timer was running slow, with each 'second" taking place over two real seconds. It'll update by 2 seconds to compensate. No more absurdly long timers! also added animal cruelty.

**Comment by burn1cycle** [comment]
Is it a simple or complex effort to incorporate the computer records database entered crimes?

 I often enter crimes in on the Sec HUD or Computer Records and genpop has no interaction with this and it would be nice to have it transfer over quickly with a wave of the soon-to-be prisoner's ID(their role issued ID) onto the Prisoner Management Interface and have a prisoner ID come out if the crimes were set in the Database(or if capital crime erroring them to check with HoS/Captain). This way I don't have to do it twice.

YOU DONT KNOW YOUR FUCKING SEC TIME #12387 is a problem I have too.
Adjacent to the above, it would be nice if it printed the station time that the prisoner gets out in and even says "You feel so bored you do the math in your head and its XX:XX minutes/seconds until you get out".

Other issue I have is the single panel glass isn't as strong as full tile and can be more easily broken through fire, pressure, and brute. Can we get a buff in the numbers to put them on par with full tile reinforced glass? 

I have liked the improvements to brigging a lot. It's much easier and feels a lot less archaic. Seriously our old brig system was OLD, I don't think it changed in years and it felt like it it was a stiff and clunky piece of crud. New system is sleek and shiny.

**Comment by maxion12345** [comment]
> Is it a simple or complex effort to incorporate the computer records database entered crimes?
> 
> I often enter crimes in on the Sec HUD or Computer Records and genpop has no interaction with this and it would be nice to have it transfer over quickly with a wave of the soon-to-be prisoner's ID(their role issued ID) onto the Prisoner Management Interface and have a prisoner ID come out if the crimes were set in the Database(or if capital crime erroring them to check with HoS/Captain). This way I don't have to do it twice.
> 
> YOU DONT KNOW YOUR FUCKING SEC TIME #12387 is a problem I have too. Adjacent to the above, it would be nice if it printed the station time that the prisoner gets out in and even says "You feel so bored you do the math in your head and its XX:XX minutes/seconds until you get out".
> 
> Other issue I have is the single panel glass isn't as strong as full tile and can be more easily broken through fire, pressure, and brute. Can we get a buff in the numbers to put them on par with full tile reinforced glass?
> 
> I have liked the improvements to brigging a lot. It's much easier and feels a lot less archaic. Seriously our old brig system was OLD, I don't think it changed in years and it felt like it it was a stiff and clunky piece of crud. New system is sleek and shiny.

As long as the prisoners name is properly set, it'll automatically add crimes to their sec record. In regards to the ID issue I've not been able to reproduce either on main or private server, so I'm not sure if it's an unrelated bug or not. Maybe it's byond, dunno. 

I think most of the issues were handled with the change from a big cell back to three small ones. The ID part gets people confused at time, and I am considering seeing if I can apply the code from this to the old timers to give it similar functionality without all the fumbling with a special prisoner ID.

**Comment by maxion12345** [comment]
Closing in favor of https://github.com/yogstation13/Yogstation/pull/12422

may re-open if testmerge for that one goes badly

**Pull request closed without merging**
